### PR TITLE
Disable passing of sm_arch to iree-compile CL args by default.

### PR DESCRIPTION
**Pull request by monorimet**
This commit fixes numerics issues caused by TensorFloat32 execution on TensorFlow Masked LM models and PyTorch MobileBert.

TODO:
Re-enable the passing of this flag for when models require benchmarking -- this is a little sticky as we first validate, then use the same compiled .vmfb to benchmark on pytest runs.

Perhaps we can relax the tolerance when --benchmark is passed.

### CODE REVIEW

File: shark/parser.py [hunk 1]
```diff
@@ -47,16 +47,10 @@ def dir_file(path):
     default="./shark_tmp",
 )
 parser.add_argument(
-    "--save_mlir",
+    "--enable_tf32",
```
Comment (powderluv) [inline shark/parser.py:50]: adjusting tolerance based on this flag would be cool

**Review by powderluv — APPROVED** [review]

**Comment by powderluv** [comment]
I assume we still set the target sm_arch right ? just that tf32 is disabled without the flag ?
